http_backend.go

```diff
@@ -1,11 +1,16 @@
 package colly
 
 import (
+	"crypto/sha1"
```

Comment:
sha1 considered unsafe, please use sha256 instead.
(https://www.schneier.com/blog/archives/2017/02/sha-1_collision.html)

Reply:
sha1 is used here as a convenient way of serializing URL address. Safety is not that crucial here. If somebody can make sha1 collision in purpose, then he probably has many other ways of making web scrapping problematic. Main feature I use from the checksum it's random distribution of first 2 letters (which are used to create a directory).
I made a benchmark (using openssl) and it looks like sha1 is even faster than md5:
```
type             16 bytes     64 bytes    256 bytes   1024 bytes   8192 bytes
md5              61097.08k   175900.80k   377240.92k   518029.27k   603403.61k
sha1             74219.17k   208662.02k   485175.72k   726962.18k   846761.59k
sha256           68712.16k   155092.59k   290635.95k   358352.90k   398336.00k
```
I am also aware of possible sha1 collisions, but do you really think that it can be a case here?

Reply:
You're right, sha1 is totally enough in this case.